doc/modules/grid_search.rst

```diff
@@ -192,11 +192,53 @@ result in an error when using multiple metrics.
 See :ref:`sphx_glr_auto_examples_model_selection_plot_multi_metric_evaluation.py`
 for an example usage.
 
+.. _composite_grid_search:
+
 Composite estimators and parameter spaces
 -----------------------------------------
+`GridSearchCV` and `RandomizedSearchCV` allow searching over parameters of
+composite or nested estimators such as `Pipeline`, `ColumnTransformer`,
+`VotingClasssifier` or `CalibratedClassifierCV`
+using a dedicated syntax ``<estimator>__<parameter>`` syntax::
+
+  >>> from sklearn.model_selection import GridSearchCV
+  >>> from sklearn.calibration import CalibratedClassifierCV
+  >>> from sklearn.ensemble import RandomForestClassifier
+  >>> from sklearn.datasets import make_moons
+  >>> X, y = make_moons()
+  >>> calibrated_forest = CalibratedClassifierCV(
+  ...    base_estimator=RandomForestClassifier(n_estimators=10))
+  >>> param_grid = {
+  ...    'base_estimator__max_depth': [2, 4, 6, 8]}
+  >>> search = GridSearchCV(calibrated_forest, param_grid, cv=5)
+  >>> search.fit(X, y)
+  GridSearchCV(cv=5,
+               estimator=CalibratedClassifierCV(base_estimator=RandomForestClassifier(n_estimators=10)),
+               param_grid={'base_estimator__max_depth': [2, 4, 6, 8]})
+
+Here, ``<estimator>`` is the parameter name of the nested estimator,
+in this case ``base_estimator``.
+The `Pipeline` class has a slightly different notation, as explained in :ref:`pipeline`,
```

Comment:
It's not Pipeline alone... ColumnTransformer, VotingClassifier, etc, do this too... It's more like "If, like a Pipeline, the meta-estimator is constructed as a collection of other estimators, ..."

The other thing that's confusing about this is that, according to `get_params`, the step names *are* parameter names of the Pipeline. They're just not constructor params...